### PR TITLE
fix(wyh): marl dist loss

### PR DESCRIPTION
**Pull request by Weiyuhong-1998**
## Description
Dist loss support MARL

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/rl_utils/tests/test_td.py [hunk 1]
```diff
@@ -132,6 +132,46 @@ def test_dist_nstep_td():
...
+    batch_size = 4
+    action_dim = 3
+    agent_num = 2
+    n_atom = 51
+    v_min = -10.0
+    v_max = 10.0
+    nstep = 5
+    dist = torch.randn(batch_size, agent_num, action_dim, n_atom).abs().requires_grad_(True)
+    next_n_dist = torch.randn(batch_size, agent_num, action_dim, n_atom).abs()
+    done = torch.randn(batch_size)
```
Comment (PaParaZz1) [inline ding/rl_utils/tests/test_td.py:146]: prepare proper `done` data

File: ding/rl_utils/tests/test_td.py [hunk 2]
```diff
@@ -132,6 +132,46 @@ def test_dist_nstep_td():
...
+    )
+    reward = torch.randn(nstep, batch_size)
+    data = dist_nstep_td_data(dist, next_n_dist, action, next_action, reward, done, None)
+    loss, _ = dist_nstep_td_error(data, 0.95, v_min, v_max, n_atom, nstep)
+    assert loss.shape == ()
+    assert dist.grad is None
+    loss.backward()
+    assert isinstance(dist.grad, torch.Tensor)
+    weight = torch.tensor([0.9])
+    value_gamma = torch.tensor(0.9)
```
Comment (PaParaZz1) [inline ding/rl_utils/tests/test_td.py:167]: use python scalar 0.9

File: ding/rl_utils/tests/test_td.py [hunk 3]
```diff
@@ -216,6 +256,38 @@ def test_dist_1step_compatible():
     assert pytest.approx(nstep_loss.item(), onestep_loss.item())
 
 
+@pytest.mark.unittest
+def test_dist_1step_multi_agent_td():
```
Comment (PaParaZz1) [inline ding/rl_utils/tests/test_td.py:260]: validate the following two cases are equivalent:
- call `dist_1step_td_error` with `(B, A, *)` shape one time
- call `dist_1step_td_error` with `(B, *)` shape `agent` times

File: ding/rl_utils/td.py [hunk 4]
```diff
@@ -117,19 +117,36 @@ def dist_1step_td_error(
 ) -> torch.Tensor:
     dist, next_dist, act, next_act, reward, done, weight = data
```
Comment (PaParaZz1) [inline ding/rl_utils/td.py:118]: add overview comments to describe two cases (single-agent, multi-agent)

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

**Review by PaParaZz1 — APPROVED** [review]
